tests/functional/flows/test_update_flow.py

```diff
@@ -0,0 +1,162 @@
+
+    # Check values for simple fields
+    simple_fields = {
+        "Owner": pool.get_assigned_usernames("owner")[0],
+        "Title": title or "",
+        "Subtitle": subtitle or "",
+        "Description": description or "",
+    }
+
+    for name, value in simple_fields.items():
+        assert value_for_field_from_output(name, result.output) == value
+
+    # Check all multi-value fields
+    expected_sets = {
+        "Keywords": set(keywords),
+        "Administrators": {
+            *[
+                principal
+                for principal in SPECIAL_PRINCIPALS
+                if principal in flow_administrators
+            ],
+            *pool.get_assigned_usernames("administrators"),
+        },
+        "Starters": {
+            *[
+                principal
+                for principal in SPECIAL_PRINCIPALS
+                if principal in flow_starters
+            ],
+            *pool.get_assigned_usernames("starters"),
+        },
+        "Viewers": {
+            *[
+                principal
+                for principal in SPECIAL_PRINCIPALS
+                if principal in flow_viewers
+            ],
+            *pool.get_assigned_usernames("viewers"),
+        },
+    }
+
+    for name, expected_values in expected_sets.items():
+        match_list = set(value_for_field_from_output(name, result.output).split(","))
+        assert match_list == expected_values
+
+
+@pytest.mark.parametrize("name", ("administrators", "starters", "viewers", "keywords"))
+def test_empty_strings_for_csv_options(name, run_line):
+    """Verify that empty strings are converted to empty lists in the request body."""
+
+    flow_id = load_response("flows.update_flow").metadata["flow_id"]
+    # "--format json" prevents Auth calls that would need to be mocked.
+    command = f'globus flows update {flow_id} --{name} "" --format json'
+    run_line(command)
+
+    request = get_last_request()
+    key = name if name == "keywords" else f"flow_{name}"
+    assert json.loads(request.body)[key] == []
+
+
+def test_omitted_options(run_line):
+    """Verify that omitted CSV options are not included in the request body."""
+
+    flow_id = load_response("flows.update_flow").metadata["flow_id"]
+
+    # "--format json" prevents Auth calls that would need to be mocked.
+    command = f"globus flows update {flow_id} --format json"
+    run_line(command)
+
+    request = get_last_request()
+    omitted_keys = {"flow_administrators", "flow_starters", "flow_viewers", "keywords"}
+    assert json.loads(request.body).keys() & omitted_keys == set()
+
+
+@pytest.mark.parametrize("option", ["definition", "input-schema"])
+def test_json_object_option_validation(option, run_line):
+    """Ensure options that must be JSON objects are validated as such."""
+
+    command = f"globus flows update {uuid.uuid4()} --{option} '[]'"
+    run_line(command, assert_exit_code=2, matcher="must be a JSON object")
```

Comment:
I wish that `matcher` did this! But it does something a bit weirder/clumsier:
https://github.com/globus/globus-cli/blob/3e27e695631bc5f0122901f02c4130f66cfcb37f/tests/conftest.py#L275

I would be onboard to change the interface to support this usage (may I suggest we follow `pytest`'s `match=...` arg and support regexes?), and find some other way of handling the cases which use the current `matcher=True` behavior.

For now though, let's make this capture output and do an `assert ... in output`?

Reply:
This is resolved now. Thanks for catching this!